### PR TITLE
Fix type signature for `Integer.parse` to get correct widget for `radix`

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

Small fix based on issues found during Advent of Code.

Before:
image

After:
image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

**Review by AdRiley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso [hunk 1]
```diff
@@ -1287,7 +1287,7 @@ type Integer
          Parse the text "20220216" into an integer number.
 
             Integer.parse "20220216"
-    parse text:Text (radix=10:Integer) -> Integer ! Number_Parse_Error = integer_parse text radix
+    parse text:Text (radix:Integer = 10) -> Integer ! Number_Parse_Error = integer_parse text radix
```
Comment (JaroslavTulach) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:1290]: Isn't this a flaw of the Enso language?

Personally I never know whether to type the former or the latter variant or some other. I try few and whichever compiles I keep. I wasn't aware there could be a difference between `r=10:Int` and `r:Int=10`!

Does there have to be a difference? CCing @kazcw.

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:1290]: I think the behaviour is correct.

`radix:Integer = 10` is an argument `radix` with type ascription `Integer` and default value `10`.

`radix=10:Integer` is an argument `radix` without type ascription, with default value computed by an expression `10:Integer` - so it is the literal constant `10` plus a type-ascription that asserts that that literal is an integer.

The latter is equivalent to `radix=my_default_radix` where
```
my_default_radix =
    c = 10
    c : Integer
```

Comment (JaroslavTulach) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:1290]: Yes, that's how it behaves at present. However we could also parse `radix=10:Integer` as
- `radix` argument name
- `10` default expression value
- and type ascription `Integer` 

and interpret it as `radix:Integer = 10` probably (if the AST isn't against it completely). Possibly @kazcw could even unify these two cases for us, so we don't have to change anything in the `TreeToIr`.

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:1290]: We _could_ but I don't know of any language that allows to reorder types and default arguments like this. Won't that make the language unnecessarily complicated?

Even if we change the parser like you suggest, we still have the case of `radix=(10:Integer)`. With the parentheses I think we still should treat it as bound to the expression, not the argument, right?

**Pull request merged**